lite2/store/db/db_test.go

```diff
@@ -26,7 +28,7 @@ func TestLast_FirstSignedHeaderHeight(t *testing.T) {
 
 	// 1 key
 	err = dbStore.SaveSignedHeaderAndValidatorSet(
-		&types.SignedHeader{Header: &types.Header{Height: 1}}, &types.ValidatorSet{})
+		&types.SignedHeader{Header: &types.Header{Height: 1, ProposerAddress: tmrand.Bytes(20)}}, vals)
```

Comment:
```suggestion
		&types.SignedHeader{Header: &types.Header{Height: 1, ProposerAddress: tmrand.Bytes(crypto.AddressSize)}}, vals)
```

Reply:
why the need to fill ProposerAddress?

Reply:
when we go back and forth from proto we call validatebasic which checks proposerAddr is not len(0)

Reply:
what about other fields then? we aren't checking them?

Reply:
we call ValidateHash in which we check if its > 0 && == tmhash.Size.that is why other fields pass

Reply:
hmm... okay.. we should probably align these too. either we require all OR not. In the latter case, in validate we can add `len(proposeraddress) ==0 ||`